app/validators/content_item_uniqueness_validator.rb

```diff
@@ -29,16 +30,15 @@ def validate(record)
     base_path = location.base_path
     user_version = user_facing_version.number
 
-    matching_items = ContentItemFilter.filter(
+    other_content_items = ContentItem.where("content_items.id <> #{content_item.id}")
+    matching_items = ContentItemFilter.new(scope: other_content_items).filter(
       state: state_name,
       locale: locale,
       base_path: base_path,
       user_version: user_version,
     )
 
-    additional_items = matching_items - [content_item]
-
-    if additional_items.any?
+    if matching_items.any?
```

Comment:
I wondered if there was another micro-optimisation if we used `.count > 0` here rather than `.any?`.  I thought that `.any?` would load all the records from the DB and use ruby stdlib `Enumerable#any?` which would be slow if there were many records, but it turns out that's not what it does as `ActiveRecord::Relation#any?` basically does `ActiveRecord::Relation#count` anyway (see: https://github.com/rails/rails/blob/52ce6ece8c8f74064bb64e0a0b1ddd83092718e1/activerecord/lib/active_record/relation.rb#L297-L300 and https://github.com/rails/rails/blob/52ce6ece8c8f74064bb64e0a0b1ddd83092718e1/activerecord/lib/active_record/relation.rb#L278-L288).

TIL &c.